src/test/java/com/productservice/controllers/ProductControllerTest.java

```diff
@@ -0,0 +1,162 @@
+
+		JSONArray responce = new JSONArray();
+		responce.put(mockProdJson);
+
+		RequestBuilder requestBuilder = MockMvcRequestBuilders.get("/products").accept(MediaType.APPLICATION_JSON);
+
+		MvcResult result = mockmvc.perform(requestBuilder).andReturn();
+
+		JSONAssert.assertEquals(responce.toString(), result.getResponse().getContentAsString(), true);
+	}
+
+	@Test
+	public void getProductById() throws Exception {
+
+		Mockito.when(productService.getProduct((1L))).thenReturn(prodList.stream().findFirst());
+
+		JSONArray responce = new JSONArray();
+		responce.put(mockProdJson);
+
+		RequestBuilder requestBuilder = MockMvcRequestBuilders.get("/products/1").accept(MediaType.APPLICATION_JSON);
+
+		MvcResult result = mockmvc.perform(requestBuilder).andReturn();
+
+		JSONAssert.assertEquals(responce.get(0).toString(), result.getResponse().getContentAsString(), true);
+	}
+
+	@Test
+	public void saveProducts() throws Exception {
+
+		Mockito.when(productService.saveorUpdate(Mockito.any())).thenReturn(mockProd2);
+
+		JSONObject suceessResponse = new JSONObject();
+		suceessResponse.accumulate("id", 2);
+		suceessResponse.accumulate("name", "Samsung Full HD");
+		suceessResponse.accumulate("description", "Full HD");
+		suceessResponse.accumulate("price", 39457.5);
+		suceessResponse.accumulate("quantity", 4);
+		suceessResponse.accumulate("version", "1");
+		RequestBuilder requestBuilder = MockMvcRequestBuilders.post("/products").content(suceessResponse.toString())
+				.contentType(MediaType.APPLICATION_JSON).accept(MediaType.APPLICATION_JSON);
+
+		MvcResult result = mockmvc.perform(requestBuilder).andReturn();
+
+		JSONAssert.assertEquals(suceessResponse.toString(), result.getResponse().getContentAsString(), true);
+	}
+
+	@Test
+	public void updateProducts() throws Exception {
+
+		Mockito.when(productService.saveorUpdate(mockProd1)).thenReturn(mockProd1);
+
+		JSONObject suceessResponse = new JSONObject();
+		suceessResponse.accumulate("id", 1);
+		suceessResponse.accumulate("name", "Samsung HD");
+		suceessResponse.accumulate("description", "Default HD");
+		suceessResponse.accumulate("price", 32457.50);
+		suceessResponse.accumulate("quantity", 5);
+		suceessResponse.accumulate("version", "1");
+		RequestBuilder requestBuilder = MockMvcRequestBuilders.put("/products/1")
+				.content(suceessResponse.toString()).contentType(MediaType.APPLICATION_JSON).accept(MediaType.APPLICATION_JSON);
+
+		MvcResult result = mockmvc.perform(requestBuilder).andReturn();
+
+		assertEquals(suceessResponse.toString(), result.getResponse().getContentAsString());
+	}
+
+	@Test
+	public void deleteProduct() throws Exception {
+
+		Mockito.when(productService.delete(2L)).thenReturn(true);
+		RequestBuilder requestBuilder = MockMvcRequestBuilders.delete("/products/2");
+		MvcResult result = mockmvc.perform(requestBuilder).andReturn();
+		assertEquals("Product deleted sucessfully", result.getResponse().getContentAsString());

```

Comment:
Better to depend on status code for doing assertions, instead of string comparison.